integration-test/helpers/harness.js

```diff
@@ -72,5 +84,41 @@ export async function setup (ops = {}) {
         return server
     }
 
-    return { browser, teardown, setupServer }
+    /**
+     * A wrapper around page.goto() that supports sending additional
+     * arguments to content-scope's init methods + waits for a known
+     * indicators to avoid race conditions
+     *
+     * @param {import("puppeteer").Page} page
+     * @param {string} urlString
+     * @param {Record<string, any>} [args]
+     * @returns {Promise<void>}
+     */
+    async function gotoAndWait (page, urlString, args = {}) {
+        const url = new URL(urlString)
+
+        // Append the flag so that the script knows to wait for incoming args.
+        url.searchParams.append('wait-for-init-args', 'true')
+
+        await page.goto(url.href)
+
+        // wait until contentScopeFeatures.load() has completed
+        await page.evaluate(() => {
+            return window.__content_scope_status === 'loaded'
```

Comment:
Shouldn't these evaluates be waitForFunction?

Reply:
Nope we don't need it, removed

Reply:
yep! I had moved to `evaluate` because you need to if your return value is a promise (which mine was, but isn't any more) - pushing the change :)